string[] names
int64[] values
